src/MongoDB/Query.c

```diff
@@ -50,64 +50,244 @@ PHONGO_API zend_class_entry *php_phongo_query_ce;
 
 zend_object_handlers php_phongo_handler_query;
 
-/* {{{ proto void Query::__construct(array|object $filter[, array $options = array()])
-   Constructs a new Query */
-PHP_METHOD(Query, __construct)
+/* Appends a string field into the BSON options. Returns true on
+ * success; otherwise, false is returned and an exception is thrown. */
+static bool php_phongo_query_opts_append_string(bson_t *opts, const char *opts_key, zval *zarr, const char *zarr_key TSRMLS_DC)
 {
-	php_phongo_query_t       *intern;
-	zend_error_handling       error_handling;
-	zval                     *zfilter;
-	zval                     *zoptions = NULL;
-	bson_t                    bfilter;
-	bson_t                    boptions = BSON_INITIALIZER;
-	SUPPRESS_UNUSED_WARNING(return_value_ptr) SUPPRESS_UNUSED_WARNING(return_value) SUPPRESS_UNUSED_WARNING(return_value_used)
+	zval *value = php_array_fetch(zarr, zarr_key);
 
+	if (Z_TYPE_P(value) != IS_STRING) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"%s\" %s to be string, %s given", zarr_key, zarr_key[0] == '$' ? "modifier" : "option", zend_get_type_by_const(Z_TYPE_P(value)));
```

Comment:
Do you have tests for this case? (All `php_phongo_query_opts_append_*` functions)